### PR TITLE
fix coprocessor metrics collect bug.

### PR DESCRIPTION
**Pull request by hicqu**
There was a bug when collecting coprocessor metrics. The bug would
cause TiKV print wrong logs and return coprocessor Response with
bad `exec_detail`.

### CODE REVIEW

**Comment by hicqu** [comment]
@AndreMouche @breeswish PTAL thanks.

**Comment by breeswish** [comment]
LGTM

File: src/coprocessor/dag/executor/table_scan.rs [hunk 1]
```diff
@@ -201,10 +201,10 @@ impl Executor for TableScanExecutor {
     }
 
     fn collect_metrics_into(&mut self, metrics: &mut ExecutorMetrics) {
-        metrics.merge(&mut self.metrics);
         if let Some(scanner) = self.scanner.as_mut() {
             scanner.collect_statistics_into(&mut metrics.cf_stats);
         }
+        metrics.merge(&mut self.metrics);
```
Comment (AndreMouche) [inline src/coprocessor/dag/executor/table_scan.rs]: What's the difference?

**Pull request merged**